pkg/impersonation/impersonation.go

```diff
@@ -29,34 +30,39 @@ func New(user user.Info, clusterContext *config.UserContext) Impersonator {
 	return Impersonator{user: user, clusterContext: clusterContext}
 }
 
-func (i *Impersonator) SetUpImpersonation() (*corev1.ServiceAccount, error) {
-	sa, err := i.checkServiceAccount()
+func (i *Impersonator) SetUpImpersonation() error {
+	role, err := i.checkAndUpdateRole()
```

Comment:
Can we create the rules here since we know we need them in either case? That saves us from creating them twice when we have to fall through to create

Reply:
done